### PR TITLE
feat: making the agglayer-prover runnable

### LINKED ISSUE #276: AggLayer Prover can be started
As an **AggLayer CLI**, I want to allow admin to start an `AggLayer Prover`, so that it can receive new queries from `AggLayer` node.

### Pending questions and comments

- [ ] What configuration is needed for the prover ?
- [ ] Should we add metrics directly ?

### Linked tickets

- [x] #274 
- [ ] #286
- [ ] #287

### Definition of Done
- [ ] All acceptance criteria are met.
- [ ] Code is reviewed and approved.
- [ ] Necessary tests are written and pass.
- [ ] Documentation is updated, if applicable.
- [ ] Feature is part of a release (created or planned).

### LINKED ISSUE #287: Add and expose metrics for the `AggLayer Prover`

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR is adding everything necessary to make the agglayer prover runnable.

Fixes #276
Fixes #287 

## PR Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-telemetry/src/lib.rs [hunk 1]
```diff
@@ -59,6 +59,36 @@ lazy_static! {
...
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_request_recv")
+                .with_description("Number of proving request received")
+                .init();
+        pub static ref PROVING_REQUEST_SUCCEED: opentelemetry::metrics::Counter<u64> =
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_request_succeed")
+                .with_description("Number of proving request that succeed")
+                .init();
+        pub static ref PROVING_REQUEST_FAILED: opentelemetry::metrics::Counter<u64> =
```
Comment (hadjiszs) [inline crates/agglayer-telemetry/src/lib.rs:79]:
```suggestion
        pub static ref PROVING_REQUEST_SUCCEEDED: opentelemetry::metrics::Counter<u64> =
            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
                .u64_counter("proving_request_succeeded")
                .with_description("Number of proving request that succeeded")
                .init();
        pub static ref PROVING_REQUEST_FAILED: opentelemetry::metrics::Counter<u64> =
```

**Review by hadjiszs — APPROVED** [review]

File: crates/agglayer-prover/src/lib.rs [hunk 2]
```diff
@@ -1 +1,92 @@
...
+                    info!("Received SIGINT (ctrl-c), shutting down...");
+                    // Cancel the global cancellation token to start the shutdown process.
+                    global_cancellation_token.cancel();
+                    // Wait for the node to shutdown.
+                    node.await_shutdown().await;
+                    // Wait for the metrics server to shutdown.
+                    _ = metrics_handle.await;
+                }
+            }
+        });
```
Comment (iljakuklic) [inline crates/agglayer-prover/src/lib.rs:86]: Does the cancellation token communication work correctly if used across runtimes like this? Looks slightly suspicious to me.

File: crates/agglayer-prover/src/prover.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        health_reporter
+            .set_serving::<ProofGenerationServiceServer<ProverRPC>>()
+            .await;
+
+        health_reporter
+            .set_serving::<ProofGenerationServiceServer<ProverRPC>>()
+            .await;
+
+        let reflexion = tonic_reflection::server::Builder::configure()
```
Comment (iljakuklic) [inline crates/agglayer-prover/src/prover.rs:50]:
```suggestion
        let reflection = tonic_reflection::server::Builder::configure()
```

File: crates/agglayer-prover/src/lib.rs (hunk 2 above)
Comment (Freyskeyd) [reply crates/agglayer-prover/src/lib.rs:86]: Yes, it is the whole point of using a `CancellationToken` as it relies on `node_tree` internally and notify every waker and waiters on each future in each runtime.

**Review by iljakuklic — APPROVED** [review]

**Pull request merged**